engine/common/oncall_gateway/utils.py

```diff
@@ -59,3 +64,92 @@ def create_slack_connector(oncall_org_id: str, slack_id: str, backend: str):
+    """
+    client = ChatopsProxyAPIClient(settings.ONCALL_GATEWAY_URL, settings.ONCALL_GATEWAY_API_TOKEN)
+    try:
+        client.register_tenant(service_tenant_id, cluster_slug, SERVICE_TYPE_ONCALL)
+    except Exception as e:
+        logger.error(
+            f"create_oncall_connector: failed " f"oncall_org_id={service_tenant_id} backend={cluster_slug} exc={e}"
+        )
+        register_oncall_tenant_async.apply_async(
+            kwargs={
+                "service_tenant_id": service_tenant_id,
+                "cluster_slug": cluster_slug,
+                "service_type": SERVICE_TYPE_ONCALL,
+            },
+            countdown=2,
+        )
+
+
+def unregister_oncall_tenant(service_tenant_id: str, cluster_slug: str):
+    """
+    unregister_oncall_tenant unregisters tenant asynchronously.
+    """
+    unregister_oncall_tenant_async.apply_async(
+        kwargs={
+            "service_tenant_id": service_tenant_id,
+            "cluster_slug": cluster_slug,
+            "service_type": SERVICE_TYPE_ONCALL,
+        },
+        countdown=2,
+    )
+
+
+def can_link_slack_team(
+    service_tenant_id: str,
+    slack_team_id: str,
+    cluster_slug: str,
+) -> bool:
+    """
+    can_link_slack_team checks if it's possible to link slack workspace to oncall tenant located in cluster.
+    All oncall tenants linked to same slack team should have same cluster.
+    """
+    client = ChatopsProxyAPIClient(settings.ONCALL_GATEWAY_URL, settings.ONCALL_GATEWAY_API_TOKEN)
+    try:
+        response = client.can_slack_link(service_tenant_id, cluster_slug, slack_team_id, SERVICE_TYPE_ONCALL)
+        return response.status_code == 200
+    except Exception as e:
+        logger.error(
+            f"can_link_slack_team: slack installation impossible: {e} "
+            f"service_tenant_id={service_tenant_id} slack_team_id={slack_team_id} cluster_slug={cluster_slug}"
+        )
+
+        return False
+
+
+def link_slack_team(service_tenant_id: str, slack_team_id: str):
+    client = ChatopsProxyAPIClient(settings.ONCALL_GATEWAY_URL, settings.ONCALL_GATEWAY_API_TOKEN)
+    try:
+        client.link_slack_team(service_tenant_id, slack_team_id, SERVICE_TYPE_ONCALL)
+    except Exception as e:
+        logger.error(
+            f'msg="Failed to link slack team: {e}"'
+            f"service_tenant_id={service_tenant_id} slack_team_id={slack_team_id}"
+        )
+        link_slack_team_async.apply_async(
+            kwargs={
+                "service_tenant_id": service_tenant_id,
+                "slack_team_id": slack_team_id,
+                "service_type": SERVICE_TYPE_ONCALL,
+            },
+            countdown=2,
+        )
+
+
+def unlink_slack_team(service_tenant_id: str, slack_team_id: str):
+    unlink_slack_team_async.apply_async(
+        kwargs={
+            "service_tenant_id": service_tenant_id,
+            "slack_team_id": slack_team_id,
+            "service_type": SERVICE_TYPE_ONCALL,
+        }
+    )
```

Comment:
nit: having a function that just calls a celery task seems a bit sus to me, maybe it makes sense to call the task directly in caller code?

Reply:
The idea is that callers just use utils from the gateway package, without knowing anything about sync/async calls